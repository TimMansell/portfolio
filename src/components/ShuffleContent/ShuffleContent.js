import React from 'react';
import PropTypes from 'prop-types';

export class ShuffleContent extends React.Component {
    constructor(props) {
		super(props);

        this.shuffleContent = React.Children.toArray(props.children);

        this.state = {
            content: this.shuffleContent[0]
        };
	}

    componentDidMount() {
        this.tick = setInterval(this.rotateContent, 4000);
    }

    componentWillUnmount() {
        window.clearInterval(this.tick);
    }

    rotateContent = () => {
        this.shuffleContent.push(this.shuffleContent.shift());

        this.setState(prevState => ({
            content: this.shuffleContent[0]
        })); 
    }

    render() {
        const { content } = this.state;

        return <div>{content}</div>;
    }
}

ShuffleContent.propTypes = {
	children: PropTypes.node.isRequired
};

export default ShuffleContent;