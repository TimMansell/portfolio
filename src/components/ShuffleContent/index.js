import ShuffleContent from './ShuffleContent';

export default ShuffleContent;
