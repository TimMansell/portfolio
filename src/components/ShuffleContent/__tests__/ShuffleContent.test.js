import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ShuffleContent from '../ShuffleContent';

describe('ShuffleContent', () => {
  it('should render my component', () => {
    const props = {
      children: '<p></p>',
    };

    // eslint-disable-next-line
    const wrapper = shallow(<ShuffleContent {...props}/>);
  });

  it('should match snapshot', () => {
    const props = {
      children: '<p>Text</p>',
    };
    const snapshot = renderer.create(<ShuffleContent {...props} />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
