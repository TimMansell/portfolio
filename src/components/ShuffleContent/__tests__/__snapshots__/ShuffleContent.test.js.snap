// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShuffleContent should match snapshot 1`] = `
<div>
  &lt;p&gt;Text&lt;/p&gt;
</div>
`;
